### PR TITLE
s3_bucket: fix condition checking of requester_pays param.

### PR DESCRIPTION
**Pull request by r0ckpine**
##### SUMMARY

Fix get_bucket_request_payment error handling.

Related: [#49676](https://github.com/ansible/ansible/issues/49676)
The fix against above issue [#49628](https://github.com/ansible/ansible/pull/49628) seems not to be completed.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

s3_bucket

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/s3_bucket.py
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/s3_bucket.py:250]: Do you mind updating the documentation too, that the module doesn't fail if requester_pays is False and isn't implemented:
https://github.com/ansible/ansible/blob/1f73eb63c42470b8356658750c85088765bcfca1/lib/ansible/modules/cloud/amazon/s3_bucket.py#L105-L108

Comment (klemmster) [reply lib/ansible/modules/cloud/amazon/s3_bucket.py:250]: The root cause: since requester_pays defaults to `False` it can never be `None`.
Just stumbled over that as well.

**Comment by klemmster** [comment]
This is also present in 2.8.6

File: lib/ansible/modules/cloud/amazon/s3_bucket.py
Comment (r0ckpine) [reply lib/ansible/modules/cloud/amazon/s3_bucket.py:250]: Thanks the comment, I updated the module doc and push it again.

**Comment by r0ckpine** [comment]
Could you review the patch again? @s-hertel @klemmster

**Review by klemmster — APPROVED** [review]
Looks good to me.
I'd really appreciate if that could be merged, so we can stop patching our local ansible installations.

**Review by jillr — APPROVED** [review]
rebuild_merge

**Comment by klemmster** [comment]
What's the process here? Anything I can do now?

**Comment by tremble** [comment]
LGTM

Sorry about that, it's hit and miss if the bot does quite what we expect (it generally does what it's told, but we don't always fully understand its rules).  If it doesn't automerge this time I'll follow up with one of the core devs to get this merged.

**Comment by tremble** [comment]
bot_status

**Comment by jillr** [comment]
Sorry about that, just kicked off a fresh CI run to clear the `stale_ci` and will merge when it comes back.

**Pull request merged**
